docs: add microshift podman desktop example blog

cdrage opened the pull request:
docs: add microshift podman desktop example blog

### What does this PR do?

* Creates a bootc blog using microshift and podman desktop
* Blog covers how to create a microshift container with RHEL / Red Hat
  tools and then create it all via Podman Desktop

### Screenshot / video of UI

<!-- If this PR is changing UI, please include
screenshots or screencasts showing the difference -->

N/A

### What issues does this PR fix or reference?

<!-- Include any related issues from Podman Desktop
repository (or from another issue tracker). -->

Closes https://github.com/containers/podman-desktop-extension-bootc/issues/816

### How to test this PR?

<!-- Please explain steps to verify the functionality,
do not forget to provide unit/component tests -->

Follow the steps.

N/A on unit tests

Signed-off-by: Charlie Drage <charlie@charliedrage.com>

dgolovin commented on website/blog/2024-09-19-bootc-microshift.md:
```diff
@@ -0,0 +1,284 @@
+
+Before starting the tutorial, ensure you have:
+
+- An active Red Hat account
```
Would be good to have a link to registration form for those who does not have one.

vrothberg replied:
Maybe link to https://developers.redhat.com/articles/2024/05/07/podman-desktop-red-hat-developer-subscription?

dgolovin commented on website/blog/2024-09-19-bootc-microshift.md:
```diff
@@ -0,0 +1,284 @@
+
+- An active Red Hat account
+- Access to an [OpenShift Hybrid Cloud pull secret](https://console.redhat.com/openshift/install/pull-secret)
+- [Podman Desktop installed](https://podman-desktop.io/downloads)
```
There should be 'Podman VM is running' item or login would not work.

cdrage replied:
on linux we do not need VM running so I just left it up to the user to make sure podman desktop is installed, etc.

slemeur commented:
cc @vrothberg to review the content

vrothberg reviewed (commented):
LGTM, nice work!

Did you loop in MicroShift folks?

cdrage commented:
> LGTM, nice work!
> 
> Did you loop in MicroShift folks?

Pinged them on slack!

ggiguash commented on website/blog/2024-09-19-bootc-microshift.md:
```diff
@@ -0,0 +1,284 @@
+- Access to an [OpenShift Hybrid Cloud pull secret](https://console.redhat.com/openshift/install/pull-secret)
+- [Podman Desktop installed](https://podman-desktop.io/downloads)
+- [Podman Desktop BootC Extension](https://github.com/containers/podman-desktop-extension-bootc)
+- Your preferred VM-running software (e.g., [using libvirt](https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/8/html/configuring_and_managing_virtualization/getting-started-with-virtualization-in-rhel-8_configuring-and-managing-virtualization#enabling-virtualization-in-rhel8_virt-getting-started) or [qemu](https://www.qemu.org/))
```
We should probably point to RHEL 9 docs here.

cdrage replied:
@ggiguash can you provide a link if possible?

ggiguash replied:
@cdrage , does [this](https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/9/html/configuring_and_managing_virtualization/introducing-virtualization-in-rhel_configuring-and-managing-virtualization#introducing-virtualization-in-rhel_configuring-and-managing-virtualization) work?

cdrage replied:
Ahhh got it! thank you.

ggiguash commented on website/blog/2024-09-19-bootc-microshift.md:
```diff
@@ -0,0 +1,284 @@
+  echo 'AuthorizedKeysFile /usr/ssh/%u.keys .ssh/authorized_keys .ssh/authorized_keys2' >> /etc/ssh/sshd_config.d/30-auth-system.conf && \
+  echo ${SSH_PUBLIC_KEY} > /usr/ssh/root.keys && chmod 0600 /usr/ssh/root.keys
+
+# For simplicity, we echo the repo location for to /etc/yum.repos.d/minishift.repo
+RUN echo -e '# OpenShift 4.17 release\n\
+[ocp]\n\
+name=OpenShift 4.17\n\
+baseurl=https://mirror.openshift.com/pub/openshift-v4/$basearch/microshift/ocp/latest-4.17/el9/os/\n\
+enabled=1\n\
+gpgcheck=0\n\
+\n\
+# Dependencies\n\
+[ocp-deps]\n\
+name=OpenShift 4.17 dependencies\n\
+baseurl=https://mirror.openshift.com/pub/openshift-v4/$basearch/dependencies/rpms/4.17-el9-beta/\n\
+enabled=1\n\
+gpgcheck=0' > /etc/yum.repos.d/minishift.repo
```
Is there a reason we're pointing to pre-release MicroShift repositories? 
I think we should install a released version of MicroShift as described [here](https://github.com/openshift/microshift/blob/main/docs/contributor/image_mode.md?plain=1#L39-L45).

ggiguash commented on website/blog/2024-09-19-bootc-microshift.md:
```diff
@@ -0,0 +1,284 @@
+gpgcheck=0' > /etc/yum.repos.d/minishift.repo
+
+# Install MicroShift
+RUN dnf install -y microshift
+
+# Enable MicroShift on boot
+RUN systemctl enable microshift
```
These layers better be squashed. See the previous comment.

ggiguash commented on website/blog/2024-09-19-bootc-microshift.md:
```diff
@@ -0,0 +1,284 @@
+RUN systemctl enable microshift
+
+# For MicroShift to access the required containers to pull we will provide our OpenShift Pull Secret from https://console.redhat.com/openshift/install/pull-secret
+ARG OPENSHIFT_PULL_SECRET
+RUN echo $OPENSHIFT_PULL_SECRET > /etc/crio/openshift-pull-secret
+RUN chmod 600 /etc/crio/openshift-pull-secret
```
I'm not sure we should recommend embedding private secrets into container images (or VMs).
Can we set up the pull secret at runtime as described [here](https://github.com/openshift/microshift/blob/main/docs/contributor/image_mode.md?plain=1#L229)?

ggiguash commented on website/blog/2024-09-19-bootc-microshift.md:
```diff
@@ -0,0 +1,284 @@
+- Your public SSH key (if available) so you may remotely access your build
+- An [OpenShift Hybrid Cloud pull secret](https://console.redhat.com/openshift/install/pull-secret)
+
+Pass these as `SSH_PUBLIC_KEY` and `OPENSHIFT_PULL_SECRET` in the build page arguments.
```
We should not pass secrets at build time and store them in the image.

ggiguash commented on website/blog/2024-09-19-bootc-microshift.md:
```suggestion
$ ssh localhost -p 2222
```

ggiguash commented on website/blog/2024-09-19-bootc-microshift.md:
```suggestion
$ ssh localhost -p 2222
```

ggiguash commented on website/blog/2024-09-19-bootc-microshift.md:
```suggestion
$ ssh localhost -p 2222
```

cdrage commented:
I have updated the PR:

- Steps for copying over the secret is now added, as to not include the secret when building
- Updates to the password / username (we login with redhat/pass now)
- Use the Containerfile from the official microshift docs which is accessible via downloading by curl / clicking the link.

@ggiguash

ggiguash commented on website/blog/2024-09-19-bootc-microshift.md:
Note that MicroShift does not have storage in this configuration because it requires an LVM partition with free space. This is one of the limitations of using the RAW images.

I'm not sure if you'd like to ignore it in the tutorial, give a note, or solve the problem.

cdrage commented on website/blog/2024-09-19-bootc-microshift.md:
```diff
@@ -0,0 +1,296 @@
+hide_table_of_contents: false
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
```
![Screenshot 2024-10-28 at 9 14 10 AM](https://github.com/user-attachments/assets/61a940ac-dec1-4bfb-8912-2df09bc44e8c)

You have to click on the "Netlify" link in the Details for our CI to see it rendered :)

cdrage commented on website/blog/2024-09-19-bootc-microshift.md:
```diff
@@ -0,0 +1,296 @@
+
+Install the BootC Podman Desktop extension from the extensions catalog:
+
+![install](img/bootc-microshift/install.png)
```
part of the catalog and unable to remove

ggiguash commented on website/blog/2024-09-19-bootc-microshift.md:
The problem is not in the limitation of MicroShift storage driver. It's because you chose to use raw partition. When booting from DVD for example, it's possible to configure storage using kickstart as described [here](https://github.com/openshift/microshift/blob/main/docs/contributor/image_mode.md?plain=1#L262) for example.

ggiguash commented:
@cdrage , this is LGTM for me.

slemeur commented on website/blog/2024-09-19-bootc-microshift.md:
copying the file, or the content of the file?
I'm worry that here, it's going to erase an existing kube/config setup.

Shouldn't we point to:
`kubectl config set-cluster microshift` ...

slemeur reviewed (changes requested):
Overall good job! 
A couple of questions and minor feedback

cdrage commented:
Rebasing & renaming file to today's date and setting to auto-merge!
